Azure.ResourceManager.Core/Resources/Location.cs

```diff
@@ -2,98 +2,445 @@
+
+            return new Location(other);
         }
 
-        public static implicit operator Location(string other)
+        /// <summary>
+        /// Creates a string implicitly from a Location object.
+        /// </summary>
+        /// <param name="other">Location object to be assigned.</param>
+        public static implicit operator string(Location other) => other.DisplayName;
+
+        /// <summary>
+        /// Detects if a location object is equal to another location instance or a string representing the location name.
+        /// </summary>
+        /// <param name="other">Location object or name as a string.</param>
+        /// <returns>True or false.</returns>
+        public bool Equals(Location other)
         {
-            return new Location(other);
+            if (other == null)
+                return false;
+
+            return Name == other.Name && CanonicalName == other.CanonicalName && DisplayName == other.DisplayName;
+        }
+
+        /// <summary>
+        /// Gets the display name of a location object.
+        /// </summary>
+        /// <returns>Display name.</returns>
+        public override string ToString()
+        {
+            return DisplayName;
         }
 
-        /// <inheritdoc cref="IComparable{Location}" />
+        /// <summary>
+        /// Compares this Location name to another Location to expose if it is greater, less or equal than this one.
+        /// </summary>
+        /// <param name="other">Location object or name as a string.</param>
+        /// <returns>-1 for less than, 0 for equals, 1 for greater than.</returns>
         public int CompareTo(Location other)
         {
             if (ReferenceEquals(other, null))
             {
                 return 1;
             }
 
-            return string.Compare(Name, other.Name);
+            return Name.CompareTo(other.Name);
         }
 
-        public int CompareTo(string other)
+        private static string NormalizationUtility(string value)
         {
-            if (ReferenceEquals(other, null))
+            if (string.IsNullOrEmpty(value))
             {
-                return 1;
+                return value;
             }
 
-            return Name.CompareTo(other);
-        }
+            var sb = new StringBuilder(value.Length);
+            for (var index = 0; index < value.Length; ++index)
+            {
+                var c = value[index];
+                if (char.IsLetterOrDigit(c))
+                {
+                    sb.Append(c);
+                }
+            }
 
-        public bool Equals(Location other)
-        {
-            return CanonicalName == other.CanonicalName;
+            return sb.ToString().ToUpperInvariant();
         }
 
-        public bool Equals(string other)
+        private static NameType DetectNameType(string location)
         {
-            return CanonicalName == GetCanonicalName(other);
+            string canonicalPattern = "^[a-z]+(-[a-z]+)+(-[1-9])?$";
```

Comment:
Can we make these two regex static readonly fields?

Reply:
No, it says that the static and readonly keywords are not valid for those items.

Another possible way to do it is to declare them at the class level, would that be the way to go?

Reply:
yeah thats what i mean, fields means class variable which is where the private static readonly modifiers apply

Reply:
Got it